BuildAutomation/BuildCloudbaseInitSetup.ps1

```diff
@@ -76,6 +84,54 @@ try
 
     $python_template_dir = join-path $cloudbaseInitInstallerDir "Python$($pythonversion.replace('.', ''))_${platform}_Template"
 
+    ExecRetry -maxRetryCount 3 {
+        $pythonVersionInstaller = $pythonversion.Replace("-",".").Replace("_",".").Trim()
+        $pythonVersionInstallerSuffix = $pythonVersionInstaller
+        if ($pythonversionPrelease) {
+            $pythonVersionInstallerSuffix = $pythonversionPrelease
+        }
+        $pythonArchInstaller = ""
+        if ($platform -eq "x64") {
+            $pythonArchInstaller = "-amd64"
+        }
+        $pythonInstallerName = "python-installer-${pythonVersionInstallerSuffix}-${platform}.exe"
+        $pythonInstallerPath = (Join-Path $pwd $pythonInstallerName)
+        $PythonInstallerUrl = "https://www.python.org/ftp/python/${pythonVersionInstaller}/python-${pythonVersionInstallerSuffix}${pythonArchInstaller}.exe"
+        DownloadFile $PythonInstallerUrl $pythonInstallerPath
+        $expectedSha1Hash = $PythonInstallerSha1Hash[$pythonInstallerName]
+        if (!$expectedSha1Hash) {
+            throw "expected Sha1 Hash for ${pythonInstallerPath} is not configured"
+        }
+        $sha1Hash  = (Get-FileHash -Algorithm SHA1 $pythonInstallerPath).Hash
+        if ($sha1Hash -ne $expectedSha1Hash) {
+            throw "$pythonInstallerPath SHA1 hash is: ${sha1Hash}. Expected hash: ${expectedSha1Hash}"
+        }
+
+        try {
+            Write-Host "Trying to uninstall Python ${pythonVersionInstaller} to ${python_template_dir}"
+            $installProcess  = Start-Process -PassThru -Wait $pythonInstallerPath `

```

Comment:
I'm a bit concerned about removing existing Python installations, it might mess someone's dev environment.

Instead of uninstalling/installing Python, can we use the "embeddable" zip archives?

e.g. https://www.python.org/ftp/python/3.12.3/python-3.12.3-embed-amd64.zip

Reply:
I was thinking about the embed version too, but it is not that straightforward. The embeded Python is a subset of Python with less features, and we will need to navigate the dependency graph to install the required parts (pip, setuptools and maybe others). You will end up making the required parts of an installer, but with extra steps and without any SHA1 verifiable start point.

Reply:
I have also made a branch with the Embedded version. There are a few things to be aware of the embedded version though as it is not suitable to build CPYTHON extensions:

 * Include folder does not exist
 * Python.lib files do not exist
 * pyconfig.h does not exist
 
The include folder can be downloaded from the github cpython repository.
Python.lib files can be built from cpython source code or gotten from the MSI installer installation directory, same with pyconfig.h.

If the wheels are already available from pypi, the Embedded version will work without these requirements.

Reply:
So by getting rid of ``netifaces``, we should be able to easily use the embeddable version. We'd have to do that anyway, since it's no longer maintained.

So we can't use ``wmi`` because the provider is not available at that stage. What we could do would be to implemented an utils module that calls the following using ctypes: https://github.com/al45tair/netifaces/blob/53fcdb6e5dccc84f6734939cfee1a95d3f470d7b/netifaces.c#L1304